add new plants, remove incorrectly named plants, and add some synonym…

bleds22e opened the pull request:
…s--all based on herbarium records

gmyenni commented:
Can you add the details in a comment here? (eg submitted as caly wrig, id'd as rafi neom)

bleds22e commented:
**Added:**
Submitted as _Sarcostemma sp._, identified as _Funastrum hartwegii_
Submitted as _Cymopterus multivernatus_, identified as _Ambrosia confertifola_
Submitted as _unknown shrub_, identified as _Ambrosia salsola_
Submitted as caly wrig, identified as _Rafinesquia neomexicana_
Submitted as sida spin, identified as _Sida neomexicana_
Larrea tridenta, creosote, added to list

**Removed:**
Submitted as bout grac, identified as bout barb

**Replaced:**
Submitted as _Malvella lepidota_, replaced with _Abutilon parvulum_
Submitted as _Lupinus sparsiflorus_, replaced with _Lupinus brevicaulis_

**Name Updates:**
mach tenr: sp name updated from tanacetifolia to tagetina
erio lemm: sp name updated from lemmonii to acuminata
